src/main/frontend/pipeline-console-view/pipeline-console/main/PipelineConsole.tsx

```diff
@@ -174,9 +173,14 @@ export class PipelineConsole extends React.Component<
       .then((res) => res.json())
       .then((result) => {
         console.debug("Updating stages");
-        this.setState({
-          stages: result.data.stages,
-        });
+        this.setState(
+          {
+            stages: result.data.stages,
+          },
+          () => {
+            this.handleUrlParams();
```

Comment:
Using the optional callback argument to `setState` allows `handleUrlParams` to be called after the state is actually updated. Unfortunately `handleUrlParams` is now called twice because we can't rely on the order of `setStages` or `setSteps` completing. There is likely a better way to wait on both calls to `setState` completing.